libdrgn/symbol.c

```diff
@@ -40,6 +45,39 @@ void drgn_symbol_from_elf(const char *name, uint64_t address,
 		ret->kind = DRGN_SYMBOL_KIND_UNKNOWN;
 }
 
+LIBDRGN_PUBLIC void drgn_symbol_source(struct drgn_symbol *sym,
```

Comment:
This API has no way of showing whether it succeeded or failed. Return `struct drgn_error *` and return NULL on success.

```suggestion
LIBDRGN_PUBLIC struct drgn_error *
drgn_symbol_source(struct drgn_symbol *sym,
```